libs/full/collectives/include/hpx/collectives/gather.hpp

```diff
@@ -408,6 +408,114 @@ namespace hpx::collectives {
                                 this_site, generation, root_site),
             HPX_FORWARD(T, local_result), this_site);
     }
+
+    template <typename T>
+    std::vector<T> flatten_vector(communicator fid, std::vector<std::vector<T>>&& dimensional_vector)
+    {
+        std::vector<std::vector<T>> non_flat_vector = std::move(dimensional_vector);
+        std::vector<T> current_local_result;
+        size_t totalSize = 0;
+        for (const auto& row : non_flat_vector) {
+            totalSize += row.size();
+        }
+        current_local_result.reserve(totalSize);
+        for (auto& row : non_flat_vector) {
+            current_local_result.insert(current_local_result.end(), std::make_move_iterator(row.begin()), std::make_move_iterator(row.end()));
+        }
+        return std::move(current_local_result);
```

Comment:
Never use `std::move` on a local variable being returned from a function. This disables RVO in the compiler.